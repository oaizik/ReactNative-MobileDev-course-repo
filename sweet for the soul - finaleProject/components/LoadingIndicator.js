import React from 'react';
import { StyleSheet, Text, View, ActivityIndicator } from 'react-native';

function LoadingIndicator(props) {
  const { text } = props;

  return (
    <View styles={styles.indicator}>
      <ActivityIndicator size={100} color="#8b0000" />
      <Text style={styles.indicatorText}> {text} </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  indicator: {
    marginTop: 130
  },
  indicatorText: {
    textAlign: 'center',
    marginTop: 40,
    fontSize: 30
  }
});

export default LoadingIndicator