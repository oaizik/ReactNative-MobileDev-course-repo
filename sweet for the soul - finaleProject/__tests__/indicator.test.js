import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import LoadingIndicator from '../components/LoadingIndicator';

describe('loading indicator component tetst', function () {
  test('render loading indicator component with given text', function () {
    const { getByText } = render(<LoadingIndicator text={'indicator text...'} />);
    expect(getByText('indicator text...')).toBeDefined();
  });

  test('render loading indicator component snapshot', function () {
    const { baseElement } = render(<LoadingIndicator text={'indicator text...'} />);
    expect(baseElement).toMatchSnapshot();
  });
});