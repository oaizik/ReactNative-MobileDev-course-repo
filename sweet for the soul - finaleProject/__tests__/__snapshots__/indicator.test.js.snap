// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading indicator component tetst render loading indicator component snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      styles={
        Object {
          "marginTop": 130,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#8b0000"
        hidesWhenStopped={true}
        size={100}
      />
      <Text
        style={
          Object {
            "fontSize": 30,
            "marginTop": 40,
            "textAlign": "center",
          }
        }
      >
         
        indicator text...
         
      </Text>
    </View>
  </View>
</View>
`;
